producer-js/browser-based-ingestion/library/src/putMediaCall.js

```diff
@@ -0,0 +1,45 @@
+const AWS = require('aws-sdk')
+const AWS4 = require('aws4')
+
+const putMedia = async function (videoFile, service, region, accessKeyID, secretAccessKey, sessionToken, dataEndpoint, streamName) {
+  const dataEndpointPutMedia = new AWS.Endpoint(dataEndpoint)
+  const url = dataEndpointPutMedia.href + 'putMedia'
+
+  const options = AWS4.sign(
+    {
+      method: 'POST',
+      host: dataEndpointPutMedia.host,
+      path: '/putMedia',
+      service: service,
+      region: region,
+      headers: {
+        Accept: '*/*',
+        'x-amzn-fragment-timecode-type': 'RELATIVE',
+        'x-amzn-producer-start-timestamp': Date.now() / 1000,
+        'x-amzn-stream-name': streamName
+      }
+    },
+    {
+      accessKeyId: accessKeyID,
+      secretAccessKey: secretAccessKey,
+      sessionToken: sessionToken
+    }
+  )
+  options.body = videoFile
+  const request = new Request(url, options)
+  fetch(request)
```

Comment:
Since, you're using async, it's better to not chain the response with then, the flow is much better if you use await instead. Also, remove logging the response and the error should be thrown as an error.

```javascript
const response = await fetch(request);
const text = response.text();

if (data.includes('"EventType":"ERROR"')) {
  throw new Error('The following error has occurred:\n' + data + '\nReload the page and try again')
}
```